File: java/src/test/java/com/example/routeauthorizationtesting/RouteAuthSpecs.java
package com.example.routeauthorizationtesting;

import com.example.routeauthorizationtesting.Access.AnyRole;
import com.example.routeauthorizationtesting.Access.Unauthenticated;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;

import java.util.Set;

import static org.springframework.http.HttpMethod.*;

@Configuration
class ConfigureRouteAuthSpecs {
    @Bean
    Set<RouteAuthSpec> configure() {
        return Set.of(
            new RouteAuthSpec("/user/{id}", GET, new AnyRole(Role.ADMIN, Role.BASIC)),
            new RouteAuthSpec("/user/{id}", PUT, new AnyRole(Role.ADMIN, Role.BASIC)),
            new RouteAuthSpec("/user/{id}", PATCH, new AnyRole(Role.ADMIN, Role.BASIC)),
            new RouteAuthSpec("/user", POST, new AnyRole(Role.ADMIN)),
            new RouteAuthSpec("/user/{id}", DELETE, new AnyRole(Role.ADMIN)),
            new RouteAuthSpec("/some-route-without-correct-auth-to-show-failure", POST, new AnyRole(Role.ADMIN))
        );
    }
}

File: kotlin/src/test/kotlin/com/example/routeauthorizationtesting/RouteAuthSpecs.kt
package com.example.routeauthorizationtesting

import com.example.routeauthorizationtesting.Access.AnyRole
import com.example.routeauthorizationtesting.Access.Unauthenticated
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.springframework.http.HttpMethod.*

typealias RouteAuthSpecs = Set<RouteAuthSpec>

@Configuration
class ConfigureRouteAuthSpecs {
    @Bean
    fun configure(): RouteAuthSpecs = setOf(
        RouteAuthSpec("/user/{id}", GET, AnyRole(Role.ADMIN, Role.BASIC)),
        RouteAuthSpec("/user/{id}", PUT, AnyRole(Role.ADMIN, Role.BASIC)),
        RouteAuthSpec("/user/{id}", PATCH, AnyRole(Role.ADMIN, Role.BASIC)),
        RouteAuthSpec("/user", POST, AnyRole(Role.ADMIN)),
        RouteAuthSpec("/user/{id}", DELETE, AnyRole(Role.ADMIN)),
        RouteAuthSpec("/some-route-without-correct-auth-to-show-failure", POST, AnyRole(Role.ADMIN)),
        RouteAuthSpec("/actuator/info", GET, Unauthenticated),
        RouteAuthSpec("/actuator/health", GET, Unauthenticated)
    )
}
